lib/fileOperations.js

```diff
@@ -196,11 +198,55 @@ function writeFile(path, data, options, callback) {
   }
 }
 
+function mkdirp(dirpath, mode, callback) {
+  if (typeof mode === 'function') {
+    callback = mode;
+    mode = undefined;
+  }
+
+  var m = mode || DEFAULT_DIR_MODE;
+  var cb = callback || function() {};
+  dirpath = path.resolve(dirpath);
+
+  fs.mkdir(dirpath, m, function(er) {
```

Comment:
It doesn't seem to call `mkdir` back down the path (cb is being called, ending the cycle).  This probably shouldn't be using `mkdirp` at all because it has to chmod each path segment